src/lib/mina/actions/offchain-state.ts

```diff
@@ -47,9 +55,15 @@ type OffchainState<Config extends { [key: string]: OffchainStateKind }> = {
    *
    * This tells the offchain state about the account to fetch data from and modify, and lets it handle actions and onchain state.
    */
-  setContractInstance(
-    contract: SmartContract & { offchainState: State<OffchainStateCommitments> }
-  ): void;
+  setContractInstance(contract: OffchainStateContract): void;
+
+  /**
+   * Set the smart contract class that this offchain state is connected with.
+   *
+   * This is an alternative for `setContractInstance()` which lets you compile offchain state before knowing an contract instance.
+   * However, you must call `setContractInstance()` before calling `createSettlementProof()`.
+   */
+  setContractClass(contract: OffchainStateContractClass): void;
```

Comment:
this is needed by the testing framework (and presumably other users as well), to compile the offchain state program before having a contract instance at hand